src/test/java/uk/nhs/prm/repo/ehrtransferservice/database/TransferServiceTest.java

```diff
@@ -0,0 +1,226 @@
+
+@ExtendWith(MockitoExtension.class)
+class TransferServiceTest {
+    @Mock
+    private TransferRepository transferRepository;
+
+    @Mock
+    private ConversationActivityService activityService;
+
+    @InjectMocks
+    private TransferService transferService;
+
+    private final ConversationRecordBuilder conversationRecordBuilder = new ConversationRecordBuilder();
+
+    private static final String NHS_NUMBER = "9798547485";
+    private static final String SOURCE_GP = "B45744";
+    private static final String NEMS_MESSAGE_ID = "2d74a113-1076-4c63-91bc-e50d232b6a79";
+    private static final String DESTINATION_GP = "A74854";
+    private static final String NEMS_EVENT_LAST_UPDATED = "2023-10-09T15:38:03.291499328Z";
+
+    @Test
+    void createConversation_ValidNewConversationOrResetForRetryRequest_Ok() {
+        // given
+        final UUID inboundConversationId = UUID.randomUUID();
+        final RepoIncomingEvent repoIncomingEvent = createRepoIncomingEvent(inboundConversationId);
+
+        // when
+        when(transferRepository.isInboundConversationPresent(inboundConversationId)).thenReturn(false);
+
+        // then
+        assertDoesNotThrow(() -> transferService.createConversationOrResetForRetry(repoIncomingEvent));
+
+        verify(transferRepository).isInboundConversationPresent(inboundConversationId);
+        verify(activityService).captureConversationActivity(inboundConversationId);
+        verify(transferRepository).createConversation(repoIncomingEvent);
+
+        verify(transferRepository, never()).updateConversationStatus(any(), any());
+    }
+
+    @Test
+    void createConversation_ValidRetriedConversationOrResetForRetryRequest_Ok() {
+        // given
+        final UUID inboundConversationId = UUID.randomUUID();
+        final RepoIncomingEvent repoIncomingEvent = createRepoIncomingEvent(inboundConversationId);
+        final ConversationRecord conversationRecord = createConversationRecord(inboundConversationId, INBOUND_REQUEST_SENT);
+
+        // when
+        when(transferRepository.isInboundConversationPresent(inboundConversationId)).thenReturn(true);
+        when(transferRepository.findConversationByInboundConversationId(inboundConversationId)).thenReturn(conversationRecord);
+
+        // then
+        assertDoesNotThrow(() -> transferService.createConversationOrResetForRetry(repoIncomingEvent));
+
+        verify(transferRepository).isInboundConversationPresent(inboundConversationId);
+        verify(transferRepository).findConversationByInboundConversationId(inboundConversationId);
+        verify(activityService).captureConversationActivity(inboundConversationId);
+        verify(transferRepository).updateConversationStatus(inboundConversationId, INBOUND_STARTED);
+
+        verify(transferRepository, never()).createConversation(any());
+    }
+
+    @Test
+    void createExceptionOrResetForRetry() {

```

Comment:
This method name doesn't look right...

Reply:
Good spot. Renamed a few tests here to the proper names.